libr/socket/socket_http.c

```diff
@@ -3,45 +3,81 @@
 #include <r_socket.h>
 #include <r_util.h>
 
-static int __socket_slurp (RSocket *s, ut8 *buf, int bufsz) {
-	int i;
-	int chsz = 1;
-	// r_socket_block_time (s, 1, 1, 0);
-	if (r_socket_read_block (s, (ut8 *) buf, 1) != 1) {
+#if __WINDOWS__
+#include <WinInet.h>
+#endif
+
+static size_t __socket_slurp(RSocket *s, RBuffer *buf) {
+	size_t i;
+	ut8 c;
+	r_socket_block_time (s, 1, 1, 0);
```

Comment:
why this `r_socket_block_time` has different parameters than the one in the for-loop?

Reply:
Also, if both calls can use the same parameters you can probably just put everything inside the for-loop.